api/src/main/java/io/grpc/TlsServerCredentials.java

```diff
@@ -0,0 +1,231 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import com.google.common.io.ByteStreams;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * TLS credentials, providing server identity and encryption. Consumers of this credential must
+ * verify they understand the configuration via the {@link #incomprehensible incomprehensible()}
+ * method. Unless overridden by a {@code Feature}, server identity is provided via {@link
+ * #getCertificateChain}, {@link #getPrivateKey}, and {@link #getPrivateKeyPassword}.
+ */
+@ExperimentalApi("TODO")
+public final class TlsServerCredentials extends ServerCredentials {
+  /**
+   * Creates an instance using provided certificate chain and private key. Generally they should be
+   * PEM-encoded and the key is an unencrypted PKCS#8 key (file headers have "BEGIN CERTIFICATE" and
+   * "BEGIN PRIVATE KEY").
+   */
+  public static ServerCredentials create(File certChain, File privateKey) throws IOException {
+    return newBuilder().keyManager(certChain, privateKey).build();
+  }
+
+  /**
+   * Creates an instance using provided certificate chain and private key. Generally they should be
+   * PEM-encoded and the key is an unencrypted PKCS#8 key (file headers have "BEGIN CERTIFICATE" and
+   * "BEGIN PRIVATE KEY").
+   *
+   * <p>The streams will not be automatically closed.
+   */
+  public static ServerCredentials create(
+      InputStream certChain, InputStream privateKey) throws IOException {
+    return newBuilder().keyManager(certChain, privateKey).build();
+  }
+
+  private final boolean fakeFeature;
+  private final byte[] certificateChain;
+  private final byte[] privateKey;
+  private final String privateKeyPassword;
+
+  TlsServerCredentials(Builder builder) {
+    fakeFeature = builder.fakeFeature;
+    certificateChain = builder.certificateChain;
+    privateKey = builder.privateKey;
+    privateKeyPassword = builder.privateKeyPassword;
+  }
+
+  /**
+   * The certificate chain, as a new byte array. Generally should be PEM-encoded.
+   */
+  public byte[] getCertificateChain() {
+    return Arrays.copyOf(certificateChain, certificateChain.length);
+  }
+
+  /**
+   * The private key, as a new byte array. Generally should be in PKCS#8 format. If encrypted,
+   * {@link #getPrivateKeyPassword} is the decryption key. If unencrypted, the password must be
```

Comment:
Why does the getter have the comment "If unencrypted, the password must be null" ? Shouldn't it be on the setter side?

Reply:
Setter side already had this language. Tweaked language here to "will be" instead of "must be"